forecast_tools.R

```diff
@@ -3,12 +3,35 @@ library(lubridate)
 library(zoo)
 library(ggplot2)
 
+#Get all model aic values and calculate akaike weights
+compile_aic_weights = function(forecast_folder='./predictions'){
+  weight_filenames = list.files(forecast_folder, full.names = TRUE, recursive = TRUE)
+  weight_filenames = weight_filenames[grepl('aic_weights',weight_filenames)]
+  all_weights=data.frame()
+  
+  for(this_weight_file in weight_filenames){
+    this_weight_data = try(read.csv(this_weight_file, na.strings = '', stringsAsFactors = FALSE))
+    all_weights = all_weights %>%
+      dplyr::bind_rows(this_weight_data)
+  }
+  
+  all_weights = all_weights %>%
+    group_by(date,currency, level, species) %>%
+    mutate(delta_aic = aic-min(aic), weight = exp(-0.5*delta_aic) / sum(exp(-0.5*delta_aic))) %>%
+    ungroup()
+  return(all_weights)
+}
+
 #Create the ensemble model from all other forecasts
 #Currently just the mean of the esimates and confidence intervals.
 make_ensemble=function(all_forecasts, model_weights=NA, models_to_use=NA){
+  weights = compile_aic_weights()
+  
   ensemble = all_forecasts %>%
+    left_join(weights, by=c('date','model','currency','level','species')) %>%
     group_by(date, NewMoonNumber, forecastmonth, forecastyear,level, currency, species) %>%
-    summarise(estimate = mean(estimate), LowerPI=mean(LowerPI), UpperPI=mean(UpperPI))
+    summarise(estimate = sum(estimate*weight), LowerPI=sum(LowerPI*weight), UpperPI=sum(UpperPI*weight))
```

Comment:
For simplicity sake. I suspected there was a more statistically sound way to do it. But at the moment we don't use the CI's for much so I didn't  look into it.